packages/hooks/src/createHasHook.js

```diff
@@ -5,17 +5,24 @@
  * @param  {Object}   hooks Stored hooks, keyed by hook name.
  *
  * @return {Function}       Function that returns whether any handlers are
- *                          attached to a particular hook.
+ *                          attached to a particular hook and optional namespace.
  */
 function createHasHook( hooks ) {
 	/**
-	 * Returns how many handlers are attached for the given hook.
+	 * Returns whether any handlers are attached for the given hookName and optional namespace.
 	 *
-	 * @param  {string}  hookName The name of the hook to check for.
+	 * @param {string} hookName  The name of the hook to check for.
+	 * @param {string} namespace Optional. The unique namespace identifying the callback in the form `vendor/plugin/function`.
 	 *
 	 * @return {boolean} Whether there are handlers that are attached to the given hook.
 	 */
-	return function hasHook( hookName ) {
+	return function hasHook( hookName, namespace ) {
+		// Use the namespace if provided.
+		if ( 'undefined' !== typeof namespace ) {
+			return hookName in hooks &&
```

Comment:
If `hookName in hooks` is common to both the `true` and `false` cases of the condition, could it be extracted out as an initial test?

Additionally, in pursuit of shorter line length, we could invert the condition to allow this test to occur two tabs up:

```js
if ( ! ( hookName in hooks ) ) {
	return false;
}

if ( typeof namespace === 'undefined' ) {
	return true;
}

return hooks[ hookName ].handlers.some( ( hook ) => hook.namespace === namespace );
```

Thoughts? I could see the `return true;` being a bit more confusing than as proposed, though perhaps easy enough to clarify with an inline code comment.

Reply:
The current code feels clearer and is more concise. If the line length is a concer, can we split the line instead?